src/program.hpp

```diff
@@ -269,6 +270,10 @@ typedef struct _cl_program : public api_object {
     CHECK_RETURN cl_build_status compile_source();
     CHECK_RETURN cl_build_status link();
 
+    /// Check if all of the capabilities required by the SPIR-V module are
```

Comment:
Sorry, probably should have discussed this with you first. Not necessarily shooting for Doxygen here (though I'd definitely be in favour of doing so), but just generally documenting methods I'm adding seemed like a good idea. Good for IDE autocompletion features as well.

Reply:
No worries. I just thought you were aiming for Doxygen because of the triple forward slash. Yup, I'm all for that as long as we don't dilute the code too much.